api4/system.go

```diff
@@ -378,6 +380,18 @@ func getAnalytics(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte(rows.ToJson()))
 }
 
+func getSupportedTimezones(c *Context, w http.ResponseWriter, r *http.Request) {
+	supportedTimezones := c.App.Config().SupportedTimezones
+
+	if len(supportedTimezones) == 0 {
```

Comment:
nit: is this block doing anything that line 392 wouldn't do already?

Reply:
@lieut-data my concern was if supportedTimezones was nil, it wouldn't have the same effect as line 392.

I believed that serializing `nil` would not provide an empty JSON struct of "[]"?

Being that Go is typed, would it respect `type Supportedtimezones []string` and return the zero value for that which would be `"[]"`?

The goal is to have consistency, **always** return an array as a response.  I've had issues with other APIs where it would return null or {} to clients which expected [].

This would cause inexplicable side-effects on clients that were typed such on Android w/ Java.

Just my experience, I am very aware of Go's type system.  It could very well be that Go takes care of that for you and I did not know 😞

Reply:
Ah, my mistake -- you're right that `nil` wouldn't be handled the same, and I agree it would be best to always return an array. Perhaps just check for `!= nil` to emphasize the intent, since a a non-nil but empty `supportedTimezones` should encode the same way.

Reply:
Right, I had to step out so I wasn't able to post this example earlier: https://play.golang.org/p/qPUivJTSwYt

The reason I didn't explicitly check for `!= nil` was because len handles that.
https://golang.org/pkg/builtin/#len
> if v is nil, len(v) is zero.